pulsar-common/src/main/java/com/yahoo/pulsar/common/naming/DestinationDomain.java

```diff
@@ -16,5 +16,29 @@
 package com.yahoo.pulsar.common.naming;
 
 public enum DestinationDomain {
-    persistent
+    persistent("persistent"), non_persistent("non-persistent");
```

Comment:
@rdhabalia Can you check whether there are any problems in using `non-persistent` in a URL schema? Is the `-` a valid character there? 
If there is a problem there, we could use `nonpersistent` or `memory`